### PR TITLE
Maven project 

### PR DESCRIPTION
**Pull request by p-nedelchev**
Needs review

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,12 +16,7 @@ out/
 *.ear
 
 # User-specific stuff:
-.idea/workspace.xml
-.idea/tasks.xml
-.idea/dictionaries
-.idea/vcs.xml
-.idea/misc.xml
-.idea/jsLibraryMappings.xml
+.idea/*.xml
```
Comment (tsony-tsonev) [inline .gitignore]: you should always ignore the whole .idea directory before doing some work 
```
.idea/
```
as well as all other configuration files related to the IDE
```
*.ipr
*.iws
*.iml
```
add these lines to the `gitignore` and remove the unneeded files

**Comment by p-nedelchev** [comment]
.gitignore fixed and unnecessary files removed

**Comment by tsony-tsonev** [comment]
After adding the propper settigns to the gitignore you should:
```
git rm -r --cached *.iml
git rm -r --cached *.ipr
git rm -r --cached *.iws
git rm -r --cached .idea
git add .
git commit -m "deleting ignored files." 
```
then push the changes
these steps will remove the unwanted files from the branch

afterwards you may want to untrack some files so you don't have to revert them every time before you commit it they were somehow changed by the IDE or by compilation or etc
```shell
#git update-index --assume-unchanged <filename>
git update-index --assume-unchanged .idea/encodings.xml
```
any file will work (not sure if it will for whole directories but probably not just for files I think)

**Comment by tsony-tsonev** [comment]
In each module you should create a Class with some printi method which uses the log4j dependency lib than in the top level project you should have some main method instantiating these classes and call their printing mehtods. Then after building the project jar and executing it if all goes right you'll be sure that your dependencies are ok.

**Comment by p-nedelchev** [comment]
Done. Needs a review

**Comment by tsony-tsonev** [comment]
LGTM

**Pull request closed without merging**